cirq/google/op_deserializer.py

```diff
@@ -31,6 +36,34 @@
     import cirq
 
 
+class OpDeserializer(abc.ABC):
+    """Generic supertype for op deserializers."""
+
+    @property
+    @abc.abstractmethod
+    def serialized_id(self) -> str:
+        """Returns the string identifier for the resulting serialized object.
+
+        This value should reflect the internal_type of the serializer.
+        """
+
+    @property  # type: ignore
+    @deprecated(deadline='v0.12', fix='Use serialized_id instead.')
+    def serialized_gate_id(self) -> str:
+        return self.serialized_id
+
+    @abc.abstractmethod
+    def from_proto(
+        self,
+        proto,
+        *,
+        arg_function_language: str = '',
+        constants: List[v2.program_pb2.Constant] = None,
+        raw_constants: List[Any] = None,
```

Comment:
Do we need a docstring here?  I am not sure I understand what raw_constants is supposed to be.
If it is the deserialized version of the constants entry, I am not sure that raw_constants is a good name for it.  Perhaps deserialized_constants would be better?

Reply:
Added the docstring. `raw_constants` here is meant to mirror the argument of the same name in the serializer - would there be a better shared name for both of them?

Reply:
Resolved (renamed to `deserialized_constants` in deserializer)